gateway/internal/clients/currency/currency_service_client.go

```diff
@@ -0,0 +1,37 @@
+package currency
+
+import (
+	"context"
+
+	grpc_gen "github.com/GenesisEducationKyiv/software-engineering-school-4-0-boredgus/gateway/internal/clients/currency/gen"
+	"google.golang.org/grpc"
+)
+
+type (
+	ConvertCurrencyParams struct {
+		Base   string
+		Target []string
+	}
+
+	currencyServiceClient struct {
+		client grpc_gen.CurrencyServiceClient
+	}
+)
+
+func NewCurrencyServiceClient(conn grpc.ClientConnInterface) *currencyServiceClient {
+	return &currencyServiceClient{
+		client: grpc_gen.NewCurrencyServiceClient(conn),
+	}
+}
+
+func (c *currencyServiceClient) Convert(ctx context.Context, base string, target []string) (map[string]float64, error) {
```

Comment:
This method is a bit confusing. What are you going to convert? As I know the task was to get the currency rate. Did you add some additional functionality for converting amounts of currencies?

Reply:
This method converts base currency to target currencies (e.g. (usd, [uah]) => uahRate). it's a bit an overkill, but it supports required functionality of converting usd to uah.